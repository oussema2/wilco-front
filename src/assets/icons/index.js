/* eslint-disable global-require */
export const aircraft = require( './aircraft.png' );
export const airport = require( './airport.png' );
export const arrowLeft = require( './arrowLeft.png' );
export const arrowRight = require( './arrowRight.png' );
export const award = require( './award.png' );
export const ban = require( './ban.png' );
export const bell = require( './bell.png' );
export const binoculars = require( './binoculars.png' );
export const brandSymbol = require( './brandSymbol.png' );
export const chevronLeft = require( './chevronLeft.png' );
export const chevronRight = require( './chevronRight.png' );
export const cloud = require( './cloud.png' );
export const certificate = require( './certificate.png' );
export const chatComment = require( './chatComment.png' );
export const chatNewMessage = require( './chatNewMessage.png' );
export const checkbox = require( './checkbox.png' );
export const checkboxUnselected = require( './checkboxUnselected.png' );
export const comment = require( './comment.png' );
export const cross = require( './cross.png' );
export const eye = require( './eye.png' );
export const eyeSlash = require( './eyeSlash.png' );
export const exclamationCircle = require( './exclamationCircle.png' );
export const exclamationTriangle = require( './exclamationTriangle.png' );
export const filter = require( './filter.png' );
export const flightInfoAircraft = require( './flightInfoAircraft.png' );
export const home = require( './home.png' );
export const horizontalLogo = require( './horizontalLogo.png' );
export const infoCircle = require( './infoCircle.png' );
export const logOut = require( './logOut.png' );
export const map = require( './map.png' );
export const meatballsMenu = require( './meatballsMenu.png' );
export const megaphone = require( './megaphone.png' );
export const onBoardingFirst = require( './onBoardingFirst.png' );
export const onBoardingSecond = require( './onBoardingSecond.png' );
export const onBoardingThird = require( './onBoardingThird.png' );
export const planeDeparture = require( './planeDeparture.png' );
export const plus = require( './plus.png' );
export const photoVideo = require( './photoVideo.png' );
export const privacyOnlyMe = require( './privacyOnlyMe.png' );
export const privacyPublic = require( './privacyPublic.png' );
export const remove = require( './remove.png' );
export const route = require( './route.png' );
export const rulerVertical = require( './rulerVertical.png' );
export const star = require( './star.png' );
export const starMyFeed = require( './starMyFeed.png' );
export const search = require( './search.png' );
export const settings = require( './settings.png' );
export const sliders = require( './sliders.png' );
export const speed = require( './speed.png' );
export const stopwatch = require( './stopwatch.png' );
export const thumbsUp = require( './thumbsUp.png' );
export const user = require( './user.png' );
export const users = require( './users.png' );
export const vector = require( './vector.png' );
