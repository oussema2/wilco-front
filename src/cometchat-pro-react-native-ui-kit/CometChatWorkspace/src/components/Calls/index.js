export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
