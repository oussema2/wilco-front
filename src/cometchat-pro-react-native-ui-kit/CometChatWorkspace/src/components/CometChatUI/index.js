/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
