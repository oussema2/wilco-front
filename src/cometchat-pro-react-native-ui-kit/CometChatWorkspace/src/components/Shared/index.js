export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
