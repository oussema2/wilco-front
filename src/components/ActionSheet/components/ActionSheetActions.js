import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
