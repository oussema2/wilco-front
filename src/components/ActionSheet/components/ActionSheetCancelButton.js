import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
