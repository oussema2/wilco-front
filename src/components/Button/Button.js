import React from 'react';
import PropTypes from 'prop-types';
import { Button as ButtonElement } from 'react-native-elements';
import { TouchableOpacity } from 'react-native';
import { Image } from '../Image';
import noop from '../../helpers/noop';
import { styles } from './Button.styles';

const Button = ( {
	testID, title, buttonStyle, titleStyle, size, onPress, disabled
} ) => {
	const _sizeForButton = {
		'small': { height: 52, width: 164 },
		'medium': { height: 52, width: 236 },
		'big': { height: 60, width: 335 }
	};

	const _buttonStyle = {
		...styles.button,
		..._sizeForButton[ size ],
		...buttonStyle
	};

	const _titleStyle = {
		...styles.title,
		...titleStyle
	};

	const _iconStyles = {
		size: 24,
		...styles.icon,
		..._buttonStyle.iconStyle
	};

	const _renderIcon = () => (
		<Image
			testID="icon"
			source={_iconStyles.source}
			size={_iconStyles.size}
			tintColor={_iconStyles.tintColor}
		/>
	);

	return (
		<ButtonElement
			icon={_iconStyles.source && _renderIcon}
			testID={testID}
			buttonStyle={_buttonStyle}
			titleStyle={_titleStyle}
			title={title}
			onPress={onPress}
			disabled={disabled}
			TouchableComponent={TouchableOpacity}
		/>
	);
};

Button.propTypes = {
	title: PropTypes.string,
	testID: PropTypes.string,
	buttonStyle: PropTypes.instanceOf( Object ),
	titleStyle: PropTypes.instanceOf( Object ),
	size: PropTypes.oneOf( [ 'big', 'medium', 'small', 'custom' ] ),
	onPress: PropTypes.func,
	disabled: PropTypes.bool
};

Button.defaultProps = {
	title: '',
	testID: 'button',
	buttonStyle: {},
	titleStyle: {},
	size: 'medium',
	onPress: noop,
	disabled: false
};

export default Button;
