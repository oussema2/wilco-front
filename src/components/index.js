export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
