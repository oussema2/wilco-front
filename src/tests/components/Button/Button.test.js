import * as React from 'react';
import { render } from '@testing-library/react-native';
import { plus } from '../../../assets/icons';
import { Button } from '../../../components/Button';

describe( 'Button', () => {
	describe( 'with the default props', () => {
		it( 'renders the Button component correctly', async () => {
			const component = render( <Button /> );
			expect( component.queryByTestId( 'button' ) ).not.toBeNull();
			expect( component.getByRole( 'button' ) ).not.toBeNull();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when the size is small', () => {
		it( 'renders the small button', async () => {
			const component = render( <Button size="small" /> );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when the size is medium', () => {
		it( 'renders the medium button', async () => {
			const component = render( <Button size="medium" /> );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when the size is big', () => {
		it( 'renders the big button', async () => {
			const component = render( <Button size="big" /> );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when the size is custom', () => {
		it( 'renders the button', async () => {
			const component = render( <Button size="custom" /> );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with button styles', () => {
		it( 'renders the button correctly', async () => {
			const buttonStyle = {
				borderRadius: 2,
				backgroundColor: 'black',
				height: 10,
				width: 10
			};
			const component = render( <Button buttonStyle={buttonStyle} /> );
			expect( component ).toMatchSnapshot();
		} );

		describe( 'with an icon', () => {
			it( 'renders the icon correctly', async () => {
				const buttonStyle = {
					iconStyle: {
						source: plus,
						tintColor: 'red'
					}
				};
				const component = render( <Button buttonStyle={buttonStyle} /> );
				expect( component.queryAllByTestId( 'icon' ) ).not.toBeNull();
				expect( component ).toMatchSnapshot();
			} );
		} );
	} );
} );
