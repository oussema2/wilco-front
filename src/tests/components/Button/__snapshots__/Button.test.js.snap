// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when the size is big renders the big button 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#12A3BE",
          "borderColor": "#2089dc",
          "borderRadius": 26,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "padding": 8,
          "width": 335,
        }
      }
    />
  </View>
</View>
`;

exports[`Button when the size is custom renders the button 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#12A3BE",
          "borderColor": "#2089dc",
          "borderRadius": 26,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    />
  </View>
</View>
`;

exports[`Button when the size is medium renders the medium button 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#12A3BE",
          "borderColor": "#2089dc",
          "borderRadius": 26,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 52,
          "justifyContent": "center",
          "padding": 8,
          "width": 236,
        }
      }
    />
  </View>
</View>
`;

exports[`Button when the size is small renders the small button 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#12A3BE",
          "borderColor": "#2089dc",
          "borderRadius": 26,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 52,
          "justifyContent": "center",
          "padding": 8,
          "width": 164,
        }
      }
    />
  </View>
</View>
`;

exports[`Button with button styles renders the button correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderColor": "#2089dc",
          "borderRadius": 2,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 10,
          "justifyContent": "center",
          "padding": 8,
          "width": 10,
        }
      }
    />
  </View>
</View>
`;

exports[`Button with button styles with an icon renders the icon correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#12A3BE",
          "borderColor": "#2089dc",
          "borderRadius": 26,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 52,
          "iconStyle": Object {
            "source": Object {
              "testUri": "../../../src/assets/icons/plus.png",
            },
            "tintColor": "red",
          },
          "justifyContent": "center",
          "padding": 8,
          "width": 236,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "height": 24,
              "width": 24,
            },
          ]
        }
      >
        <FastImageView
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/assets/icons/plus.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="icon"
          tintColor="red"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Button with the default props renders the Button component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#12A3BE",
          "borderColor": "#2089dc",
          "borderRadius": 26,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 52,
          "justifyContent": "center",
          "padding": 8,
          "width": 236,
        }
      }
    />
  </View>
</View>
`;
